Classes/ViewHelpers/Asset/AbstractAssetViewHelper.php

```diff
@@ -66,65 +75,61 @@ public function render()
             return;
         }
 
-        $filePath = $this->getPublicFilePath($this->arguments['path']);
-
-        $this->addTagToPageRenderer($this->buildTag($filePath));
+        $position = $this->arguments['position'] ?: $this->defaultTagPosition;
+        $this->addResourceToPageTemplateSetup($position);
 
         $GLOBALS['SCRIPT_STYPE_PUSH_ASSETS'][] = $this->arguments['name'];
     }
 
     /**
-     * Renders the tag.
-     *
-     * @param $filePath
-     * @return string The rendered tag.
-     */
-    abstract protected function buildTag($filePath);
-
-    /**
-     * @param string $filePath
-     * @return string
+     * Add the resource to the template setup according to the given position
+     * 
+     * @param $position
      */
-    protected function getPublicFilePath($filePath)
+    protected function addResourceToPageTemplateSetup($position)
     {
-        if (!$this->arguments['external']) {
-            $absFilePath = GeneralUtility::getFileAbsFileName($filePath);
+        $key = $this->getTemplateSetupKeyForPosition($position) . '.';
 
-            if ($absFilePath) {
-                $filePath = substr($absFilePath, strlen(PATH_site));
-                $filePath = GeneralUtility::createVersionNumberedFilename($filePath);
-            }
+        // Modify page template setup
+        $setup = &$GLOBALS['TSFE']->pSetup;
+
+        if (!array_key_exists($key, $setup)) {
+            $setup[$key] = [];
         }
 
-        return $filePath;
+        $resource = $this->buildResourceInformation();
+        $resourceIndex = static::$templateSetupDataIndexCounter++;
+        $setup[$key][$resourceIndex] = $resource[0];
+        if (count($resource) > 1) {
```

Comment:
$resource has always more than one index. This check seems to be unnecessary.

Reply:
It is possible to override the `AbstractAssetViewHelper::buildResourceInformation()` in a subclass and omit the second array value.
This check prevents a possible PHP warning when accessing invalid array indexes.